### PR TITLE
[WIP] gh-92584: Rename the distutils package to _distutils

### PR DESCRIPTION
**Pull request by vstinner**

### CODE REVIEW

**Comment by vstinner** [comment]
test_tools and test_venv fail because of pip: https://github.com/pypa/pip/issues/11103

**Comment by vstinner** [comment]
By default, ``make install`` fails on ``ensurepip``, also because of pip.

Temporary workaround: ``./configure --with-ensurepip=no``: ``make install`` doesn't run ``ensurepip`` and so doesn't fail.

**Comment by vstinner** [comment]
For now, this PR is a draft, and so I didn't write any documentation (no NEWS entry, no What's New in Python 3.12, etc.).

**Comment by vstinner** [comment]
>  Docs / Docs (pull_request) Failing

``make -C Doc/ PYTHON=../python venv`` fails because of pip.

**Comment by vstinner** [comment]
Related issue: https://github.com/python/cpython/issues/92584

**Review by FFY00 — APPROVED** [review]
This is correct, and seems like the best approach. We just have to fix pip.

**Comment by vstinner** [comment]
@tiran:
> @erlend-aasland and I will have the last modules ported to Modules/Setup.stdlib very soon. We won't need distutils any more then. I recommend that we do not rename distutils and instead remove it once and for all.

See the issue https://github.com/python/cpython/issues/92584: peg_generator, c-analyzer tool and test_cppext still use distutils. I would prefer to not break them.

**Comment by vstinner** [comment]
> See the issue https://github.com/python/cpython/issues/92584: peg_generator, c-analyzer tool and test_cppext still use distutils. I would prefer to not break them.

I modified test_cppext to build the C++ extension with setuptools. So far, so good: it works on all buildbots.

We can try a similar approach for test_peg_generator. Right now, test_peg_generator builds a library to make the test faster. I don't know how to port distutils code to setuptools.

**Comment by vstinner** [comment]
@tiran:
> @erlend-aasland and I will have the last modules ported to Modules/Setup.stdlib very soon.

How is it going? Is this work tracked somewhere?

For example, configure.ac mentions the array module, but setup.py still has code to build this extension. Is there a plan to remove the setup.py code, if the configure/Makefile code is enough? Or is there still a reason to keep the setup.py code?

**Comment by tiran** [comment]
I have created GH-93939

tl;dr every module that uses ``addext()`` can be built without distutils (except for a quirk in multiprocessing). Modules that use ``add()`` are not yet ported. I have an open PR for ``_ctypes``.

**Comment by vstinner** [comment]
It seems like most core devs don't want to keep distutils hidden in the stdlib, but want to directly remove it. So I close my PR. Remaining steps are tracked by other issues.
